chainstate/src/detail/mod.rs

```diff
@@ -268,13 +260,37 @@ impl Chainstate {
         self.attempt_to_process_block(block, block_source, 0)
     }
 
+    /// Initialize chainstate with genesis block
+    pub fn process_genesis(&mut self) -> Result<(), BlockError> {
+        use chainstate_storage::{BlockchainStorageWrite, TransactionRw};
+
+        // Gather information about genesis.
+        let genesis = self.chain_config.genesis_block();
+        let genesis_id = self.chain_config.genesis_block_id();
+        let utxo_count = genesis.utxos().len() as u32;
+        let genesis_index =
+            common::chain::TxMainChainIndex::new(genesis_id.clone().into(), utxo_count)
+                .expect("Genesis not constructed correctly");
+
+        // Initialize storage with given info
+        let mut db_tx = self.chainstate_storage.transaction_rw();
+        db_tx.set_best_block_id(&genesis_id).map_err(BlockError::StorageError)?;
+        db_tx
+            .set_block_id_at_height(&BlockHeight::zero(), &genesis_id)
+            .map_err(BlockError::StorageError)?;
+        db_tx
+            .set_mainchain_tx_index(&genesis_id.into(), &genesis_index)
+            .map_err(BlockError::StorageError)?;
+        db_tx.commit().map_err(BlockError::StorageError)
+    }
```

Comment:
This is where the genesis block gets initialised and info about it is written in the storage.